aspnetcore/grpc/interceptors.md

```diff
@@ -0,0 +1,270 @@
+* `context` carries scoped-values associated with the client side call. You can use it to pass metadata like security principals, credentials or tracing data. Moreover, `context` carries information about deadlines and cancellation. For more information, see [Reliable gRPC services with deadlines and cancellation](xref:grpc/deadlines-cancellation#deadlines>).
+
+The following code shows how to add caller metadata through the context.
+
+```csharp
+private void AddCallerMetadata<TRequest, TResponse>(ref ClientInterceptorContext<TRequest, TResponse> context)
+    where TRequest : class
+    where TResponse : class
+{
+    var headers = context.Options.Headers;
+
+    // Call doesn't have a headers collection to add to.
+    // Need to create a new context with headers for the call.
+    if (headers == null)
+    {
+        headers = new Metadata();
+        var options = context.Options.WithHeaders(headers);
+        context = new ClientInterceptorContext<TRequest, TResponse>(context.Method, context.Host, options);
+    }
+
+    // Add caller metadata to call headers
+    headers.Add("caller-user", Environment.UserName);
+    headers.Add("caller-machine", Environment.MachineName);
+    headers.Add("caller-os", Environment.OSVersion.ToString());
+}
+```
+
+`LogCall`, `AddCallerMetadata` and `HandleResponse` methods are private methods embedded in the exemplary `ClientLoggerInterceptor` class. For complete implementation, see an [example of client interceptor](https://github.com/grpc/grpc-dotnet/blob/master/examples/Interceptor/Client/ClientLoggerInterceptor.cs).
+
+### Configure client interceptors
+
+gRPC client interceptors are configured by creating a call invoker from a channel and then 
+creating a client from that invoker.
+
+The following code:
+* Creates a channel by using `GrpcChannel.ForAddress`.
+* Creates an invoker by using `Intercept` method on the created channel.
+* Creates a client from the invoker.
+
+```csharp
+using var channel = GrpcChannel.ForAddress("https://localhost:5001");
+var invoker = channel.Intercept(new ClientLoggerInterceptor());
+
+var client = new Greeter.GreeterClient(invoker);
+```
+
+A call invoker intercepts the channel with the given interceptor. If you want to intercept the channel with multiple interceptors use overloaded `Intercept` method. Any number of interceptors may be used for a single RPC.
+
+```csharp
+var invoker = channel.Intercept(
+    new ClientLoggerInterceptor(),
+    new ClientMonitoringInterceptor(),
+    new ClientTokenInterceptor());
+```
```

Comment:
```suggestion
```csharp
var invoker = channel
    .Intercept(new ClientTokenInterceptor())
    .Intercept(new ClientMonitoringInterceptor())
    .Intercept(new ClientLoggerInterceptor());
```
```
IMO focus on chaining calls and mention there is a helper method for combining them